backend/src/main/java/capstone/backend/services/OrderToSupplierService.java

```diff
@@ -45,43 +50,63 @@ public OrderToSupplierDTO createOrder(OrderToSupplierDTO order) throws EntityNot
         return mapOrder(repo.save(mapOrder(order)));
     }
 
-    private boolean checkIfProductsExistent(OrderToSupplierDTO order) {
+    public OrderToSupplierDTO receiveOrder(OrderToSupplierDTO order, OrderStatus status) throws EntityNotFoundException, IllegalArgumentException {
+        if (status != RECEIVED) {
+            throw new IllegalArgumentException("We couldn't process your request!");
```

Comment:
I'd use passive voice, "Your request couldn't be processed" - otherwise, who is "we"? But of course this is totally minor ;-)

Reply:
we is the chips and circuits forming a union